pkg/autoops/api/flag_trigger_operation.go

```diff
@@ -54,36 +54,40 @@ func executeAutoOpsRuleOperation(
 	return dt.Err()
 }
 
+// If the flag is already enabled, we only print an error log.
+// Otherwise, the Operation state won't change to TRIGGERED, keeping in an infinite loop
+// trying to enable the flag.
 func enableFeature(
 	ctx context.Context,
+	ftStorage ftstorage.FeatureStorage,
 	environmentNamespace string,
-	autoOpsRule *domain.AutoOpsRule,
 	feature *ftdomain.Feature,
 	logger *zap.Logger,
 ) error {
-	req := &featureproto.EnableFeatureRequest{
-		Id:                   autoOpsRule.FeatureId,
-		Command:              &featureproto.EnableFeatureCommand{},
-		EnvironmentNamespace: environmentNamespace,
-	}
 	if err := feature.Enable(); err != nil {
 		logger.Error(
 			"Failed to enable feature flag",
 			log.FieldsFromImcomingContext(ctx).AddFields(
 				zap.Error(err),
-				zap.String("featureId", req.Id),
-				zap.String("environmentNamespace", req.EnvironmentNamespace),
+				zap.String("featureId", feature.Id),
+				zap.String("environmentNamespace", environmentNamespace),
 			)...,
 		)
+		return nil
+	}
+	if err := ftStorage.UpdateFeature(ctx, feature, environmentNamespace); err != nil {
```

Comment:
Update the flag only if the `Enabled` field changes. Otherwise, it will return a `no affected rows` error.